src/container.ts

```diff
@@ -26,54 +29,114 @@ export type ContainerEventArgs = EventArgs | WindowEventArgs | LayoutEventArgs;
  * @extends ContainerWindowManager
  * @extends ContainerNotificationManager
  */
-export interface Container extends ContainerWindowManager, ContainerNotificationManager {
+export abstract class Container extends EventEmitter implements ContainerWindowManager, ContainerNotificationManager {
```

Comment:
In order to reference static events on Container, changed from interface to class.  Moved many abstract members up from ContainerBase.